concreteproperties/stress_strain_profile.py

```diff
@@ -457,6 +457,367 @@ def __post_init__(
+                    / (1 - rho_cc)
+                )
+
+                # calculate tranverse reinforcement ratios and confining pressures across
+                # defined depth
+                rho_d = A_vd / (self.trans_spacing * b_core)
+                f_ld = k_e * rho_d * self.trans_f_y
+
+                # calculate tranverse reinforcement ratios and confining pressures across
+                # defined width
+                rho_b = A_vb / (self.trans_spacing * d_core)
+                f_lb = k_e * rho_b * self.trans_f_y
+
+                # calculate confined concrete strength
+                f_cc = self.compressive_strength * (
+                    -1.254
+                    + 2.254
+                    * (1 + 7.94 * min(f_lb, f_ld) / self.compressive_strength) ** 0.5
+                    - 2 * min(f_lb, f_ld) / self.compressive_strength
+                )
+            else:
+                # calculate core diameter
+                d_s = self.d - 2 * self.cvr - self.trans_d_b
+
+                # calculate core area
+                A_c = d_s**2 * np.pi / 4
+
+                # calculate volumetric ratio of confinement reinforcement
+                rho_st = (
+                    self.n_confinement
+                    / self.trans_spacing
+                    * (4 * self.trans_d_b**2 * np.pi / 4 / d_s)
+                )
+                # calculate ratio of reinforcement area to core area
+                rho_cc = self.long_reinf_area / A_c
+
+                # calculate confinement effectiveness coefficient
+                exp = 2 if self.sect_type in ["circ_hoop"] else 1
+                k_e = (1 - s_dash / (2 * d_s)) ** exp / (1 - rho_cc)
+
+                # calculate tranverse confining pressures
+                # rho_b = A_vb / (self.trans_spacing * d_core)
+                f_l = k_e * rho_st * self.trans_f_y
+
+                # calculate confined concrete strength
+                f_cc = self.compressive_strength * (
+                    -1.254
+                    + 2.254 * (1 + 7.94 * f_l / self.compressive_strength) ** 0.5
+                    - 2 * f_l / self.compressive_strength
+                )
+        else:
+            # calculate unconfined concrete strength
+            f_cc = self.compressive_strength
+
+        # calculate strain associated with max confined/unconfined concrete strength
+        eps_cc = self.eps_co * (1 + 5 * (f_cc / self.compressive_strength - 1))
+
+        # calculate maximum confined/unconfined compressive strain
+        if self.conc_confined:
+            eps_c_max = min(
+                0.004
+                + self.n_steel_strain * rho_st * self.trans_f_y * self.eps_su / f_cc,
+                0.05,
+            )
+        else:
+            eps_c_max = self.eps_c_max_unconfined
+
+        # calculate secant modulus
+        E_sec = f_cc / eps_cc
+
+        if self.conc_confined:
+            self.strains = np.linspace(0, eps_c_max, self.n_points)
```

Comment:
Should be of type ``List[float]`` not a numpy array, suggest adding ``.tolist()`` to end of statement.

I.e. ``self.strains = np.linspace(0, eps_c_max, self.n_points).tolist()``

Reply:
need to convert to list later on as `x = self.strains / eps_cc` throws error otherwise. Think my thinking was easier to operate on np array than add loop to calculate x.

Will update to convert to list after this calculation and amend code below this to use `.insert` & `.append` on the lists as intent was a list is returned for the stresses and strains as per other methods.

Reply:
See commit bc11ce88828df3cfcc572478fe8aa87d542ededd